File: lib/models/column_text.dart
class ColumnText{
  final String text;
final int textWidth;
  final int textAlign;
  final int textSize;


  ColumnText({
    required this.text,
    this.textWidth = 1,
    this.textAlign = 0,
    this.textSize = 28,
  });

  factory ColumnText.fromJson(Map<String, dynamic> json) {
    return ColumnText(
      text: json['text'] as String,
      textWidth: json['textWidth'] as int,
      textAlign: json['textAlign'] as int,
      textSize: json['textSize'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'text': text,
      'textWidth': textWidth,
      'textAlign': textAlign,
      'textSize': textSize,
    };
  }
}

File: android/src/main/kotlin/com/ezyorder/flutter_imin_printer/models/ColumnText.kt
package com.ezyorder.flutter_imin_printer.models

import com.squareup.moshi.Json
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types
import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory

data class ColumnText(
    val text: String,
    @Json(name = "textWidth") val textWidth: Int = 1,
    @Json(name = "textAlign") val textAlign: Int = 0,
    @Json(name = "textSize") val textSize: Int = 28,
) {

    companion object {
        fun fromJsonList(json: String): List<ColumnText>? {
            val moshi = Moshi.Builder()
                .addLast(KotlinJsonAdapterFactory())
                .build()

            val type = Types.newParameterizedType(List::class.java, ColumnText::class.java)
            val adapter = moshi.adapter<List<ColumnText>>(type)
            return adapter.fromJson(json)
        }
    }


}
